test/test_inverse.py

```diff
@@ -663,12 +667,179 @@ def test_KFAC_inverse_from_state_dict(
+)
+def test_EKFAC_inverse_exactly_damped_matmat(
+    inv_case: Tuple[
+        Module,
+        Union[MSELoss, CrossEntropyLoss],
+        List[Parameter],
+        Iterable[Tuple[torch.Tensor, torch.Tensor]],
+    ],
+    cache: bool,
+    exclude: str,
+    separate_weight_and_bias: bool,
+    delta: float = 1e-2,
+):
+    """Test matrix-matrix multiplication by an inverse (exactly) damped EKFAC approximation."""
+    model_func, loss_func, params, data, batch_size_fn = inv_case
+    dtype = torch.float64  # use double precision for better numerical stability
+    model_func = model_func.to(dtype=dtype)
+    loss_func = loss_func.to(dtype=dtype)
+    params = [p.to(dtype=dtype) for p in params]
+    data = [
+        (
+            (cast_input(x, dtype), y)
+            if isinstance(loss_func, CrossEntropyLoss)
+            else (cast_input(x, dtype), y.to(dtype=dtype))
+        )
+        for x, y in data
+    ]
+
+    if exclude is not None:
+        names = {p.data_ptr(): name for name, p in model_func.named_parameters()}
+        params = [p for p in params if exclude not in names[p.data_ptr()]]
+
+    EKFAC = EKFACLinearOperator(
+        model_func,
+        loss_func,
+        params,
+        data,
+        batch_size_fn=batch_size_fn,
+        separate_weight_and_bias=separate_weight_and_bias,
+        check_deterministic=False,
+    )
+
+    # manual exactly damped inverse
+    inv_EKFAC_naive = (
+        torch.inverse(
+            EKFAC @ torch.eye(EKFAC.shape[0], dtype=dtype, device=EKFAC._device)
+            + delta * torch.eye(EKFAC.shape[0], dtype=dtype, device=EKFAC._device)
+        )
+        .cpu()
+        .numpy()
+    )
+
+    # check that passing a tuple for exact damping will fail
+    with raises(ValueError):
+        inv_EKFAC = KFACInverseLinearOperator(
+            EKFAC, damping=(delta, delta), use_exact_damping=True
+        )
+
+    # use exact damping with KFACInverseLinearOperator
+    inv_EKFAC = KFACInverseLinearOperator(
+        EKFAC, damping=delta, cache=cache, use_exact_damping=True
+    )
+
+    num_vectors = 2
+    X = random.rand(EKFAC.shape[1], num_vectors)
+    # test for equivalence
+    report_nonclose(inv_EKFAC.to_scipy() @ X, inv_EKFAC_naive @ X)
+
+    assert inv_EKFAC._cache == cache
+    # test that the cache is empty
+    assert len(inv_EKFAC._inverse_input_covariances) == 0
+    assert len(inv_EKFAC._inverse_gradient_covariances) == 0
+
+
+def test_EKFAC_inverse_save_and_load_state_dict():
+    """Test that KFACInverseLinearOperator can be saved and loaded from state dict."""
+    torch.manual_seed(0)
+    batch_size, D_in, D_out = 4, 3, 2
+    X = torch.rand(batch_size, D_in)
+    y = torch.rand(batch_size, D_out)
+    model = torch.nn.Linear(D_in, D_out)
+
+    params = list(model.parameters())
+    # create and compute EKFAC
+    ekfac = EKFACLinearOperator(
+        model,
+        MSELoss(reduction="sum"),
```

Comment:
Any specific reason for using a non-default argument for `reduction` here?

Reply:
To make sure the reduction is correctly saved and loaded. I added a comment.